File: JANUARY/25-01-2026/1984.erl
-spec minimum_difference(Nums :: [integer()], K :: integer()) -> integer().
minimum_difference(Nums, K) when K =< 1 ->
    0;
minimum_difference(Nums, K) ->
    Sorted = lists:sort(Nums),
    Diffs = [lists:nth(I + K - 1, Sorted) - lists:nth(I, Sorted)
             || I <- lists:seq(1, length(Sorted) - K + 1)],
    lists:min(Diffs).

File: 2026POTD/JANUARY/25-01-2026/1984.swift
class Solution {
    func minimumDifference(_ nums: [Int], _ k: Int) -> Int {
        if k <= 1 { return 0 }
        let a = nums.sorted()
        var ans = Int.max
        for i in 0...a.count - k {
            ans = min(ans, a[i + k - 1] - a[i])
        }
        return ans
    }
}